src/slacko.ml

```diff
@@ -335,6 +335,64 @@ type file_obj = {
   num_stars: int option [@default None];
 } [@@deriving of_yojson { strict = false }]
 
+type field_obj = {
+  title: string option [@default None];
+  value: string option [@default None];
+  short: bool option [@default None];
+} [@@deriving yojson { strict = false }]
+
+let field ?title ?value ?short () = {
+  title;
+  value;
+  short;
+}
+
+type attachment_obj = {
+  fallback: string option [@default None];
+  color: string option [@default None];
+  pretext: string option [@default None];
+  author_name: string option [@default None];
+  author_link: string option [@default None];
+  author_icon: string option [@default None];
+  title: string option [@default None];
+  title_link: string option [@default None];
+  text: string option [@default None];
+  fields: field_obj list option [@default None];
+  image_url: string option [@default None];
+  thumb_url: string option [@default None];
+  footer: string option [@default None];
+  footer_icon: string option [@default None];
+  ts: timestamp option [@default None];
+  mrkdwn_in: string list option [@default None];
+} [@@deriving yojson { strict = false }]
+
+let attachment
+    ?fallback ?color ?pretext
+    ?author_name ?author_link ?author_icon
+    ?title ?title_link
+    ?text ?fields
+    ?image_url ?thumb_url
+    ?footer ?footer_icon
+    ?ts ?mrkdwn_in
+    () = {
+  fallback = (match fallback with Some _ -> fallback | None -> text);
```

Comment:
I put them in because the compiler thought the next field was part of the `None -> text` block, but that may have been a slightly different version of the expression. I'll check again and remove them if I can.